lib/matplotlib/animation.py

```diff
@@ -1045,22 +1079,23 @@ def func(current_frame: int, total_frames: int) -> Any
         # since GUI widgets are gone. Either need to remove extra code to
         # allow for this non-existent use case or find a way to make it work.
 
-        facecolor = savefig_kwargs.get('facecolor',
-                                       mpl.rcParams['savefig.facecolor'])
-        if facecolor == 'auto':
-            facecolor = self._fig.get_facecolor()
-
         def _pre_composite_to_white(color):
             r, g, b, a = mcolors.to_rgba(color)
             return a * np.array([r, g, b]) + 1 - a
 
-        savefig_kwargs['facecolor'] = _pre_composite_to_white(facecolor)
-        savefig_kwargs['transparent'] = False   # just to be safe!
         # canvas._is_saving = True makes the draw_event animation-starting
         # callback a no-op; canvas.manager = None prevents resizing the GUI
         # widget (both are likewise done in savefig()).
         with (writer.saving(self._fig, filename, dpi),
               cbook._setattr_cm(self._fig.canvas, _is_saving=True, manager=None)):
+            if not getattr(writer, '_supports_transparency', False):
```

Comment:
Let's not make dynamic attributes, instead add `_supports_transparency` to the writer base class.

I'm also a bit wary of using a property. This sounds like a static setting of the writer, but logically may depend on the file type and codec. I suggest to turn this into a function to communicate that it's not a pure state but inherently a computed quantity.

Reply:
I think moving it up to the ABC is reasonable, however we don't actually require that the writer passed into `save` is a sub-class of our class so we I'm +.75 on remaining defensive here and using `getattr` here.

Reply:
I beg to differ. As far as one can "require" a type in Python without explicit `isinstance` checking, we do require a subclass in
- docs: https://matplotlib.org/devdocs/api/_as_gen/matplotlib.animation.Animation.html#matplotlib.animation.Animation.save
- and type stubs https://github.com/matplotlib/matplotlib/blob/51e7230399f08788c972565fa594d7363856dffa/lib/matplotlib/animation.pyi#L165

I'd be reluctant to complicate our code just in case somebody has ducktyped a writer. But if you really want to support that, we should establish a `WriterProtocol` instead.

Reply:
I've added a `_check_isinstance`, if that's preferred.

Reply:
That's not what I meant. In Python one typically does not check the input type, but relies on behavior/capability ("the writer defines its transparency support").
This allows for dynamic duck typing, but unless explicitly documented (and e.g. codified in a Protocol), there is no guarantee on the stability of non-user facing API of such types.

In practice, this means we can introduce `_supports_transparency` and don't have to worry about compatibility. Don't jump extra hoops with getattr, just assume it's there and use it. But also don't type-check the input.